tests/end2end/features/downloads.feature

```diff
@@ -126,6 +126,14 @@ Feature: Downloading things from a website.
         Then "cancelled" should be logged
         And "cancelled" should be logged
 
+    # https://github.com/The-Compiler/qutebrowser/issues/1535
+    Scenario: Cancelling a MHTML download (issue 1535)
+        When I open data/downloads/issue1535.html
+        And I run :download --mhtml
+        And I wait 1s
```

Comment:
I see - waits are usually a bit problematic too then, because they *still* don't give you a guarantee the race condition is gone under heavy load (like on the CI), and because they make the tests a little bit slower :wink:

What I usually do in that case is to add a `And I wait for "blabla" in the log` so I force the test to fail, and then inspect the output for an appropriate thing to wait for. In this case, I added `And I wait for "fetch: PyQt5.QtCore.QUrl('http://localhost:*/drip?numbytes=128&duration=2') -> drip" in the log`.